// yfinance-live/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.yfinancedata = exports.YFinanceLive = void 0;
var WebSocket = require("isomorphic-ws");
var yfinancedata_1 = require("./yfinancedata");
Object.defineProperty(exports, "yfinancedata", { enumerable: true, get: function () { return yfinancedata_1.yfinancedata; } });
var YFinanceLive = /** @class */ (function () {
    function YFinanceLive(tickers, ondata, onclose, onerror) {
        if (!Array.isArray(tickers) ||
            !tickers.every(function (ticker) { return typeof ticker === 'string'; })) {
            throw new Error('Please provide an array of strings');
        }
        this.tickers = tickers;
        this.ws = new WebSocket('wss://streamer.finance.yahoo.com');
        this.ondata = ondata;
        this.ws.onopen = this._onopen.bind(this);
        this.ws.onmessage = this._onmessage.bind(this);
        this.ws.onclose = onclose;
        this.ws.onerror = onerror;
    }
    YFinanceLive.prototype._onopen = function (_e) {
        this.ws.send(JSON.stringify({ subscribe: this.tickers }));
    };
    YFinanceLive.prototype._onmessage = function (e) {
        var data = yfinancedata_1.yfinancedata.decode(Buffer.from(e.data, 'base64'));
        this.ondata(data);
    };
    YFinanceLive.prototype.stop = function () {
        this.ws.close();
    };
    return YFinanceLive;
}());
exports.YFinanceLive = YFinanceLive;
exports.default = (function (tickers, ondata, onclose, onerror) { return new YFinanceLive(tickers, ondata, onclose, onerror); });
